test/util.js

```diff
@@ -1,55 +1,101 @@
+     */
+    const tempDir = fs.realpathSync(os.tmpdir());
+    /**
+     * Get the pathname of the file that imported util.js.
+     * Used to create a unique directory name for each test suite.
+     */
+    const parent = parentModule();
+    const relativeParent = path.relative(process.cwd(), parent);
 
-  jest.spyOn(fsMock, 'stat').mockImplementation((path, callback) => {
-    callback(null, stats);
-  });
+    /**
+     * Each temp directory will be unique to the test file.
+     * This ensures that temp files/dirs won't cause side effects for other tests.
+     */
+    this.dir = path.resolve(tempDir, 'cosmiconfig', `${relativeParent}-dir`);
 
-  jest.spyOn(fsMock, 'statSync').mockImplementation(() => stats);
-};
+    // create directory
+    makeDir.sync(this.dir);
 
-exports.assertSearchSequence = function assertSearchSequence(
-  readFileMock,
-  searchPaths,
-  startCount
-) {
-  startCount = startCount || 0;
+    (this: any).absolutePath = this.absolutePath.bind(this);
+    (this: any).createDir = this.createDir.bind(this);
+    (this: any).createFile = this.createFile.bind(this);
+    (this: any).clean = this.clean.bind(this);
+    (this: any).deleteTempDir = this.deleteTempDir.bind(this);
+  }
 
-  expect(readFileMock).toHaveBeenCalledTimes(searchPaths.length + startCount);
+  absolutePath(dir: string) {
+    // Use path.join to ensure dir is always inside the working temp directory
+    const absolutePath = path.join(this.dir, dir);
 
-  searchPaths.forEach((searchPath, idx) => {
-    expect(readFileMock.mock.calls[idx + startCount][0]).toBe(
-      path.join(__dirname, searchPath)
-    );
-  });
-};
+    return absolutePath;
+  }
 
-function makeReadFileMockImpl(readFile) {
-  return (searchPath, encoding, callback) => {
-    try {
-      callback(null, readFile(searchPath));
-    } catch (err) {
-      callback(err);
-    }
-  };
-}
-exports.makeReadFileMockImpl = makeReadFileMockImpl;
-
-exports.mockReadFile = function mockReadFile(sync, readFile) {
-  if (sync === true) {
-    return jest.spyOn(fsMock, 'readFileSync').mockImplementation(readFile);
-  } else {
-    return jest
-      .spyOn(fsMock, 'readFile')
-      .mockImplementation(makeReadFileMockImpl(readFile));
+  createDir(dir: string) {
+    const dirname = this.absolutePath(dir);
+    makeDir.sync(dirname);
+  }
+
+  createFile(file: string, contents: string) {
+    const filePath = this.absolutePath(file);
+    const fileDir = path.parse(filePath).dir;
+    makeDir.sync(fileDir);
+
+    fs.writeFileSync(filePath, `${contents}\n`);
+  }
+
+  getSpyPathCalls(spy) {
+    const calls = spy.mock.calls;
+
+    const result = calls.map(call => {
+      const filePath = call[0];
+      const relativePath = path.relative(this.dir, filePath);
+
+      /**
+       * Replace Windows backslash directory separators with forward slashes
+       * so expected paths will be consistent cross platform
+       */
+      const normalizeCrossPlatform = relativePath.replace(/\\/g, '/');
```

Comment:
Can we use [`slash`](https://github.com/sindresorhus/slash) instead?

Reply:
I use ``slash`` in my own projects and considered adding it here, but I don't think it is worth (having to track updates etc) adding a new package for one line of code. ``slash`` has [guards against unicode and extended paths](https://github.com/sindresorhus/slash/blob/6fd50ff0bcb68812df18d526c0c3d5740fd13382/index.js#L3-L8), which isn't an issue with these tests. Do you disagree?

Also the latest version of ``slash`` doesn't support node v4 so if somehow there was an update (not sure what they could possibly change), we wouldn't benefit from it.

Reply:
Yeah, I don't think we need it unless it solves a problem we have.